xonsh/lib/os.xsh

```diff
@@ -0,0 +1,20 @@
+"""Xonsh extension of the standard library os module, using xonsh for
+subprocess calls"""
+import sys
+
+from xonsh.imphooks import install_hook
+
+install_hook
+
+import os
+from contextlib import contextmanager
+
+@contextmanager
+def indir(d):
```

Comment:
I think using `pushd` and `popd` will work better here.  Then you don't have to keep track of the "old" directory